datadog/internal/transport/custom_transport.go

```diff
@@ -85,7 +80,7 @@ func (t *CustomTransport) copyRequest(r *http.Request) *http.Request {
 	return &newRequest
 }
 
-func (t *CustomTransport) retryRequest(response *http.Response) (*time.Duration, bool) {
+func (t *CustomTransport) retryRequest(response *http.Response, retryCount int) (*time.Duration, bool) {
```

Comment:
does this code keep retrying the request if its a `5xx` or `429` forever until it works? can we make it so after like 3 retries, it terminates?

Reply:
Yes, it currently retries indefinitely until timeout period is reached or request is successful (whichever is first). Yeah i think we should be able to just cancel the context after x retries

Reply:
yeah i think this is the behavior we want (cancelling the context after x retries)